src/mixins/labels/coordinateLabels.js

```diff
@@ -0,0 +1,178 @@
+        constructor(config) {
+            super(config);
+            this.coordinateLabelsGroup = new THREE.Group();
+            this.coordinateLabelsGroup.name = COORDINATE_LABELS_GROUP_NAME;
+            this.areCoordinateLabelsShown = this.settings.areCoordinateLabelsInitiallyShown;
+            this.coordinateLabelsGroup.visible = this.areCoordinateLabelsShown;
+            this.structureGroup.add(this.coordinateLabelsGroup);
+        }
+
+        /**
+         * Formats coordinates into an array of fixed precision values
+         * @param {THREE.Vector3} position - The position vector to format
+         * @returns {Array<number>} Array of [x,y,z] coordinates with fixed precision
+         */
+        formatCoordinates(position) {
+            const precision = this.settings.roundPrecision;
+            return [
+                Number(position.x.toFixed(precision)),
+                Number(position.y.toFixed(precision)),
+                Number(position.z.toFixed(precision)),
+            ];
+        }
+
+        /**
+         * Creates a display text from coordinates
+         * @param {Array<number>} coordinates - Array of [x,y,z] coordinates
+         * @param {string} separator - Separator between coordinates
+         * @returns {string} Formatted coordinate text
+         */
+        createCoordinateText(coordinates, separator = "  ") {
+            const precision = this.settings.roundPrecision;
+            return coordinates.map((coord) => coord.toFixed(precision)).join(separator);
+        }
+
+        /**
+         * Creates a single coordinate label with proper positioning and offset
+         * @param {string} text - The coordinate text to display
+         * @param {THREE.Vector3} position - The position where to place the label
+         * @param {string} prefix - Optional prefix for the label name
+         * @returns {THREE.Sprite} The created label
+         */
+        createSingleCoordinateLabel(text, position, prefix = "coordinate") {
+            const name = `${prefix}-label-${text}`;
+            const label = this.createLabelSprite(text, name, this.settings.coordinateLabelsConfig);
+            const offsetVector = this.getCoordinateLabelOffsetVector(position, text);
+            label.userData = { atomPosition: position, atomName: text };
+            label.position.addVectors(position, offsetVector);
+            label.lookAt(this.camera.position);
+            return label;
+        }
+
+        /**
+         * Creates a hash map representing the positions (vertices) for atom labels.
+         * The hash map uses atom names as keys and corresponding 3D positions as values.
+         * If an atom name already exists in the hash map, it appends the atom's coordinates to the associated entry.
+         *
+         * @returns {Object.<string, Array.<number>>} HashMap with atom names as keys and an array of vertices as values.
+         */
+        createCoordinateVerticesHashMap() {
```

Comment:
This and the function for Elements are too repetitive.

This function should be called createVerticesHashmap(labelType), then we should have a map of processing for each label type that will include:
```js 
[
{
"labelType": "coordinate",
"processing": (atom, position) => {
                        const coordinates = this.formatCoordinates(position);
                        const text = this.createCoordinateText(coordinates);
                        return text;
    }
},
{
"labelType": "element",
"processing": (atom, position) => {
                        const text = atom.userData.symbolWithLabel;
                        return text;
    }
}
]
```